indexer/editable_map_object.cpp

```diff
@@ -70,6 +72,99 @@ bool IsProtocolSpecified(string const & website)
 {
   return GetProtocolNameLength(website) > 0;
 }
+osm::FakeNames MakeFakeSource(StringUtf8Multilang const & source,
+                              vector<int8_t> const & mwmLanguages, StringUtf8Multilang & fakeSource)
+{
+  string defaultName;
+  // Fake names works for mono language (official) speaking countries-only.
+  if (source.IsEmpty() || mwmLanguages.size() != 1 ||
+      !source.GetString(StringUtf8Multilang::kDefaultCode, defaultName))
+  {
+    return {};
+  }
+
+  osm::FakeNames fakeNames;
+  // Mwm name has higher priority then English name.
+  array<int8_t, kFakeNamesCount> fillCandidates = {{mwmLanguages.front(), StringUtf8Multilang::kEnglishCode}};
+  fakeSource = source;
+
+  string tempName;
+  for (auto const code : fillCandidates)
+  {
+    if (!source.GetString(code, tempName))
+    {
+      tempName = defaultName;
+      fakeSource.AddString(code, defaultName);
+    }
+
+    fakeNames.names.push_back({code, tempName});
```

Comment:
Is it a good idea to add tempName (that can be a default name) to the names? Looks like you have defaultName field for that.